### PR TITLE
lib/master: add unit test cases (Part I)

### PR DESCRIPTION
**Pull request by liuzix**
- Implemented mock dependencies for `BaseMaster`.
- Added two basic unit test cases for `BaseMaster`.
- Fixed some minor bugs.

### CODE REVIEW

**Comment by amyangfei** [comment]
ref #94

File: executor/server.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ type Server struct {
 
 	tcpServer   tcpserver.TCPServer
 	grpcSrv     *grpc.Server
-	cli         *client.MasterClient
+	cli         *client.MasterClientImpl
```
Comment (amyangfei) [inline executor/server.go:35]: Can we use interface `MasterClient` here?

File: lib/master.go [hunk 2]
```diff
@@ -215,6 +247,11 @@ func (m *BaseMaster) runWorkerCheck(ctx context.Context) error {
 }
 
 func (m *BaseMaster) OnError(err error) {
+	if errors.Cause(err) == context.Canceled {
+		// TODO think about how to gracefully handle cancellation here.
+		log.L().Error("BaseMaster is being canceled", zap.Error(err))
```
Comment (amyangfei) [inline lib/master.go:252]: Is warning level better here?

File: lib/common.go [hunk 3]
```diff
@@ -14,7 +14,9 @@ type (
 	WorkerStatusCode int32
 	WorkerType       int64
```
Comment (nongfushanquan) [inline lib/common.go:15]: worker type should be defined as enum

Comment (liuzix) [reply lib/common.go:15]: I will address this in another PR

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**